src/metric_model_reporter.cc

```diff
@@ -74,6 +75,25 @@ MetricReporterConfig::ParseConfig(
   // Set flag to signal to stats aggregator if caching is enabled or not
   cache_enabled_ = response_cache_enabled;
   is_decoupled_ = is_decoupled;
+
+  // Override default histogram options if set in model_metrics.
+  for (const auto& metric_control : model_metrics.metric_control()) {
+    const std::string& family_name =
+        metric_control.metric_identifier().family();
+
+    // If family name exists, override with new options.
+    if (metric_map_.find(family_name) != metric_map_.end()) {
+      // Copy protobuf RepeatedField to std::vector
+      const auto& buckets_proto = metric_control.histogram_options().buckets();
+      const prometheus::Histogram::BucketBoundaries buckets(
+          buckets_proto.begin(), buckets_proto.end());
+      histogram_options_[metric_map_.at(family_name)] = buckets;
```

Comment:
What happens if a model is dynamically reloaded/updated (--model-control-mode=poll/explicit) with different buckets after server intial load?

ex:
```
# assume modelA with some config.pbtxt that specifies some bucket values
tritonserver --model-store models --model-control-mode poll

# Change bucket values
vim models/modelA/config.pbtxt

# Triton reloads/updates model after detecting change (or manually updated in 'explicit' mode)
load_model(modelA)
```

What's the behavior? Will the buckets get updated?

I think it's OK if we don't support dynamic update in this PR - but want to know the limitations up front and make sure nothing crashes.

Reply:
Explicit mode: explicit reloading the model with new buckets will update the histogram metric.
Poll mode: dynamically updating buckets does not update the histogram metric.

I suppose two modes use different mechanisms to reload the model?